src/services/grpc/account/v1/exchange-token-for-user-data.ts

```diff
@@ -0,0 +1,62 @@
+import { config } from '@/config-manager';
+import type { GetUserDataResponse } from '@pretendonetwork/grpc/account/get_user_data_rpc';
+import type { ExchangeTokenForUserDataRequest } from '@pretendonetwork/grpc/account/exchange_token_for_user_data';
+
+export async function exchangeTokenForUserData(request: ExchangeTokenForUserDataRequest): Promise<GetUserDataResponse> {
+	if (!request.token.trim()) {
+		throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid token');
+	}
+
+	const pnid = await getPNIDByTokenAuth(request.token);
+
+	if (!pnid) {
+		throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid token');
+	}
+
+	return {
+		deleted: pnid.deleted,
+		pid: pnid.pid,
+		username: pnid.username,
+		accessLevel: pnid.access_level,
+		serverAccessLevel: pnid.server_access_level,
+		mii: {
+			name: pnid.mii.name,
+			data: pnid.mii.data,
+			url: `${config.cdn.base_url}/mii/${pnid.pid}/standard.tga`
+		},
+		creationDate: pnid.creation_date,
+		birthdate: pnid.birthdate,
+		gender: pnid.gender,
+		country: pnid.country,
+		language: pnid.language,
+		emailAddress: pnid.email.address,
+		tierName: pnid.connections.stripe.tier_name,
+		permissions: {
+			bannedAllPermanently: pnid.hasPermission(PNID_PERMISSION_FLAGS.BANNED_ALL_PERMANENTLY),
+			bannedAllTemporarily: pnid.hasPermission(PNID_PERMISSION_FLAGS.BANNED_ALL_TEMPORARILY),
+			betaAccess: pnid.hasPermission(PNID_PERMISSION_FLAGS.BETA_ACCESS),
+			accessAdminPanel: pnid.hasPermission(PNID_PERMISSION_FLAGS.ACCESS_ADMIN_PANEL),
+			createServerConfigs: pnid.hasPermission(PNID_PERMISSION_FLAGS.CREATE_SERVER_CONFIGS),
+			modifyServerConfigs: pnid.hasPermission(PNID_PERMISSION_FLAGS.MODIFY_SERVER_CONFIGS),
+			deployServer: pnid.hasPermission(PNID_PERMISSION_FLAGS.DEPLOY_SERVER),
+			modifyPnids: pnid.hasPermission(PNID_PERMISSION_FLAGS.MODIFY_PNIDS),
+			modifyNexAccounts: pnid.hasPermission(PNID_PERMISSION_FLAGS.MODIFY_NEX_ACCOUNTS),
+			modifyConsoles: pnid.hasPermission(PNID_PERMISSION_FLAGS.MODIFY_CONSOLES),
+			banPnids: pnid.hasPermission(PNID_PERMISSION_FLAGS.BAN_PNIDS),
+			banNexAccounts: pnid.hasPermission(PNID_PERMISSION_FLAGS.BAN_NEX_ACCOUNTS),
+			banConsoles: pnid.hasPermission(PNID_PERMISSION_FLAGS.BAN_CONSOLES),
+			moderateMiiverse: pnid.hasPermission(PNID_PERMISSION_FLAGS.MODERATE_MIIVERSE),
+			createApiKeys: pnid.hasPermission(PNID_PERMISSION_FLAGS.CREATE_API_KEYS),
+			createBossTasks: pnid.hasPermission(PNID_PERMISSION_FLAGS.CREATE_BOSS_TASKS),
+			updateBossTasks: pnid.hasPermission(PNID_PERMISSION_FLAGS.UPDATE_BOSS_TASKS),
+			deleteBossTasks: pnid.hasPermission(PNID_PERMISSION_FLAGS.DELETE_BOSS_TASKS),
+			uploadBossFiles: pnid.hasPermission(PNID_PERMISSION_FLAGS.UPLOAD_BOSS_FILES),
+			updateBossFiles: pnid.hasPermission(PNID_PERMISSION_FLAGS.UPDATE_BOSS_FILES),
+			deleteBossFiles: pnid.hasPermission(PNID_PERMISSION_FLAGS.DELETE_BOSS_FILES),
+			updatePnidPermissions: pnid.hasPermission(PNID_PERMISSION_FLAGS.UPDATE_PNID_PERMISSIONS)
```

Comment:
We will almost definitely want to do this again, so I think it should be put as a method on the PNID model - like `pnid.getPermissions()` or something similar

Reply:
I plan to redo the entire permissions system in the future anyway (this one is inspired by Discord, using bits, but we've decided that that's dumb). Did you want this on this PR or just add it to the task list/as part of that rework? I don't mind either way, just asking because I'm not sure how much more often we'd need code like this and this PR is holding back some Juxt changes

Reply:
We can leave it for this PR then - can you make sure that'd added to the rework task list? (I don't know where that is :P)

Reply:
I'll make an official issue for the rework, right now it's just been something we've discussed internally 👍